lib/eventFactory.js

```diff
@@ -453,50 +455,88 @@ async function convertStages(config) {
             }
         });
 
+        newStage.eventId = eventId;
         newStage.jobIds = jobIds;
+
         delete newStage.jobs; // extra field from yaml parser
 
+        // Check for setup and teardown
+        const setupJobName = `${STAGE_PREFIX}${key}:setup`;
+        const teardownJobName = `${STAGE_PREFIX}${key}:teardown`;
+        const setupJobExists = Object.keys(jobs).find(j => j === setupJobName || j === `~${setupJobName}`);
+        const teardownJobExists = Object.keys(jobs).find(j => j === teardownJobName || j === `~${teardownJobName}`);
+
+        if (setupJobExists) {
+            newStage.setup = [pipelineJobs.find(j => j.name === setupJobExists).id];
+        }
+        if (teardownJobExists) {
+            newStage.teardown = [pipelineJobs.find(j => j.name === teardownJobExists).id];
+        }
+
+        // Add workflow graph
         newStages.push(newStage);
     });
 
     return newStages;
 }
 
+/**
+ * Create stage model with stage metadata
+ * Create instance of a stage with stageId, eventId, workflowGraph, and status
+ * @method createStageBuilds
+ * @param  {Object}     config              config
+ * @param  {Number}     config.eventId      Event ID
+ * @param  {Object}     config.pipeline     Pipeline
+ * @return {Promise}
+ */
+async function createStageBuilds({ stageFactory, stageBuildFactory, stageConfig }) {
+    const stageCreateConfig = stageConfig;
+    const stageBuildCreateConfig = {
+        eventId: stageConfig.eventId,
+        workflowGraph: stageConfig.workflowGraph,
```

Comment:
`stageConfig.workflowGraph` would always be `undefined`. 

Looks like it is not being built using `stage.jobs`
https://github.com/screwdriver-cd/models/pull/570/files#diff-1c1c74750e13ea9a15f89f495aa0693768fb7ec332b4a01f9c83facdee0890deR476